### PR TITLE
Inside outside v2

### PR DESCRIPTION
**Pull request by d-robles**
Hi @kylemath this is our same compute ERP_IO plot but I added a section to the diff waves on the same axis. 

Should we merge this one with the master or keep it on insideOutsiveV2 for now?

### CODE REVIEW

**Comment by kylemath** [comment]
have you addressed all the comments about the code I made to the V1 pull request? Making it all one loop, etc. ?

**Review by kylemath — CHANGES_REQUESTED** [review]
I couldnt make line by line comments since you didn't actual make changes in this branch

So this time work on the new branch from the beginning
```
git checkout -b insideOutsideV2
```
Make all the suggested changes and tests in that branch, then commit,push, the pull request will get updated with those changes
then I can see all the changes in one place and either pull it into the master, or ask you for more changes before I do

Suggestions
-First, this is the main issue, there are actually four scripts running here, instead of everyone being analyzed in the same loop. This was a good first try at the solution, and you are close, just need to automate it,
I think the only difference between each of the four is the name of the files that you save at the end right?

they used to be named like:
```
[subs{i_sub} '_' exp '_' conds{i_cond} '_Corrected_Standard']
```

and now they are named like this:
```
[subs{i_sub} '_' exp '_facing_in_Corrected_Standard.set']
```

And also that you load in different conditions depeding on the in or out:

```
conds =  {'P_CW';'NP_CCW'}; %facing inside for regulars
conds =  {'P_CCW';'NP_CW'}; %facing OUTSIDE for regulars
```

So....
Your job as the programmer is to figure out how to make the computer do what you just did.
So combine this into one big loop for all the subjects
and make sure to rename all the P_CW and NP_CCW files as facing in
and make sure to rename all the NP_CW and P_CCW files as facing out

how do you do that?

You will likely need two things

1) is the names of those conditions as strings for the output files:
	
```conds2 = {'facing_out1'; 'facing_out2'};```

-this wasn't used after in your code, but this is the right idea, this should have a better name, like new_conds, or facing_conds, then the file naming and saving should use these

```new_conds = {'facing_in'; 'facing_out'}```

2) is a list for each subject of their goofy vs regular status
	
```
        is_goofy = [0, 0, 0, 0, 0, 0, 0, 1, ...]
	subs = {'100' '101' '102' '103' '104' '106' '107 ...'}
```

That way for each subject in the loop
if is_goofy is 0 (regular) then you can rename the files one way
if is_goofy is 1 then you can rename them the other way

Looking at the code closely these are the only things different of each of the four scripts:


```
REGULAR IN
    conds =  {'P_CW';'NP_CCW'}; %facing inside for regulars
subs = {'100' '101' '102' '103' '104' '106' '108' '111'...
  '112' '116' '117' '118' '119' '120'};
EEG = pop_saveset( EEG, 'filename',[subs{i_sub} '_' exp '_facing_in_Corrected_Standard.set'],'filepath',[Pathname 'segmentsIO\']);


REGULAR OUT
conds =  {'P_CCW';'NP_CW'}; %facing OUTSIDE for regulars
subs = {'100' '101' '102' '103' '104' '106' '108' '111'...
  '112' '116' '117' '118' '119' '120'};
EEG = pop_saveset( EEG, 'filename',[subs{i_sub} '_' exp '_facing_out_Corrected_Standard.set'],'filepath',[Pathname 'segmentsIO\']);


GOOFY IN
conds =  {'P_CCW';'NP_CW'}; %facing in - GOOFY
subs = {'107' '109' '110' '113' '114' '115' '122'};
EEG = pop_saveset( EEG, 'filename',[subs{i_sub} '_' exp '_facing_in_Corrected_Standard.set'],'filepath',[Pathname 'segmentsIO\']);


GOOFY OUT
conds =  {'P_CW';'NP_CCW'}; %facing OUT - GOOFY
subs = {'107' '109' '110' '113' '114' '115' '122'};
EEG = pop_saveset( EEG, 'filename',[subs{i_sub} '_' exp '_facing_out_Corrected_Standard.set'],'filepath',[Pathname 'segmentsIO\']);
```

***So I think your code would look something like this

```
new_conds = {'facing_in'; 'facing_out'}
conds = {'P_CW';'P_CCW'; 'NP_CW'; 'NP_CCW'}; %notice all four are here again
is_goofy = [0, 0, 0, 0, 0, 0, 0, 1, ...]
subs = {'100' '101' '102' '103' '104' '106' '107 ...'}

for i_sub = 1:length(subs)
	for i_cond = 1:length(conds)

		-load file
		-preprocess

		if is_goofy(i_sub) 
		
			%this strcmp is how to compare strings 
			% || means or, but | may be the right sympol (i always forget)

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')

				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
			else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Standard.set']
			end


		elseif ~is_goofy(i_sub) %if its a zero

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')

				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Standard.set']
				 %notice this is opposite the goofy
			else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
			end


		end

		EEG = pop_saveset( EEG, 'filename',ThisIsTheName,'filepath',[Pathname 'segmentsIO\']);

	end
end

```
I'm sure you can see how much more robust that would be, and also see how easy it would be to add the conditions clockwise vs counterclockwise into the output names if we want

**Comment by d-robles** [comment]
@kylemath I edited this part of the code and added a variable for the length of new conditions. I can't figure out why it's not saving the datasets of the goofy participants. I can continue to look at this tomorrow. 

`       if is_goofy(i_sub) 
		
			%this strcmp is how to compare strings 
			% || means or, but | may be the right sympol (i always forget)

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
            else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Standard.set']
			end


		elseif ~is_goofy(i_sub) %if its a zero

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')

				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Standard.set']
				 %notice this is opposite the goofy
			else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
            end
            
        EEG = pop_selectevent( tempEEG, 'type',3 ,'renametype','Standard','deleteevents','on','deleteepochs','on','invertepochs','off');    
        EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' new_conds{n_new} '_Corrected_Standard.set']);
        EEG = pop_saveset( EEG, 'filename',ThisIsTheName,'filepath',[Pathname 'segments_IO_V2\']);

%         EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' conds{i_cond} '_Corrected_Standard']);

        end
        
        if is_goofy(i_sub) 
		
			%this strcmp is how to compare strings 
			% || means or, but | may be the right sympol (i always forget)

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Target.set']
            else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Target.set']
			end


		elseif ~is_goofy(i_sub) %if its a zero

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')

				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Target.set']
				 %notice this is opposite the goofy
			else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Target.set']
            end
            
        EEG = pop_selectevent( tempEEG, 'type',5,'renametype','Target','deleteevents','on','deleteepochs','on','invertepochs','off');    
        EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' new_conds{n_new} '_Corrected_Target.set']);
        EEG = pop_saveset( EEG, 'filename',ThisIsTheName,'filepath',[Pathname 'segments_IO_V2\']);


		end
    end 
end`

**Comment by d-robles** [comment]
@kylemath I edited this part of the code and added a variable for the length of new conditions. I can't figure out why it's not saving the datasets of the goofy participants. I can continue to look at this tomorrow.

The code is pasting weird, I will re-paste it below

**Comment by d-robles** [comment]
`

        if is_goofy(i_sub) 
		
			%this strcmp is how to compare strings 
			% || means or, but | may be the right sympol (i always forget)

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
            else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Standard.set']
			end


		elseif ~is_goofy(i_sub) %if its a zero

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')

				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Standard.set']
				 %notice this is opposite the goofy
			else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
            end
            
        EEG = pop_selectevent( tempEEG, 'type',3 ,'renametype','Standard','deleteevents','on','deleteepochs','on','invertepochs','off');    
        EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' new_conds{n_new} '_Corrected_Standard.set']);
        EEG = pop_saveset( EEG, 'filename',ThisIsTheName,'filepath',[Pathname 'segments_IO_V2\']);

%         EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' conds{i_cond} '_Corrected_Standard']);

        end
        
        if is_goofy(i_sub) 
		
			%this strcmp is how to compare strings 
			% || means or, but | may be the right sympol (i always forget)

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Target.set']
            else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Target.set']
			end


		elseif ~is_goofy(i_sub) %if its a zero

			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')

				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Target.set']
				 %notice this is opposite the goofy
			else
				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Target.set']
            end
            
        EEG = pop_selectevent( tempEEG, 'type',5,'renametype','Target','deleteevents','on','deleteepochs','on','invertepochs','off');    
        EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' new_conds{n_new} '_Corrected_Target.set']);
        EEG = pop_saveset( EEG, 'filename',ThisIsTheName,'filepath',[Pathname 'segments_IO_V2\']);


		end
    end 
end

`

File: Skate_ERP_preprocessing_IO.m [hunk 1]
```diff
@@ -1,39 +1,30 @@
...
+    mkdir([Pathname 'segments_IO_V2\']);
 end
 [ALLEEG EEG CURRENTSET ALLCOM] = eeglab;
 
-
-for i_sub = 1:nsubs
-    for i_cond = 1:nconds
-
-                        
+for i_sub = 1:length(subs)
```
Comment (kylemath) [inline Skate_ERP_preprocessing_IO.m]: these should be nconds and nsubs not sure why you changed them

File: Skate_ERP_preprocessing_IO.m [hunk 2]
```diff
@@ -75,326 +66,77 @@
...
+			%this strcmp is how to compare strings 
+			% || means or, but | may be the right sympol (i always forget)
 
- nsubs = length(subs);
-%conds = {'P_CW';'P_CCW'; 'NP_CW'; 'NP_CCW'};
-conds =  {'P_CCW';'NP_CW'}; %facing OUTSIDE for regulars
-conds2 = {'facing_out1'; 'facing_out2'};
+			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')
+				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
+            else
```
Comment (kylemath) [inline Skate_ERP_preprocessing_IO.m:77]: from my view it looks like the indenting is not algined with the correct if statements, but I'm not sure that is the problem

File: Skate_ERP_preprocessing_IO.m [hunk 3]
```diff
@@ -75,326 +66,77 @@
...
+			if strcmp(conds{i_cond},'P_CCW') || strcmp(conds{i_cond},'NP_CW')
 
-for i_sub = 1:nsubs
-    for i_cond = 1:nconds
-          
+				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Standard.set']
+				 %notice this is opposite the goofy
+			else
+				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
+            end
```
Comment (kylemath) [inline Skate_ERP_preprocessing_IO.m:90]: make sure that this end statement matches the if is_goofy and elseif ~is_goofy

File: Skate_ERP_preprocessing_IO.m [hunk 4]
```diff
@@ -75,326 +66,77 @@
...
-%%
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-%                    GOOFY IN
-%
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-%%
-clear all
-close all
-ccc
+        if is_goofy(i_sub)
```
Comment (kylemath) [inline Skate_ERP_preprocessing_IO.m]: Why is this code repeated again down here?

File: Skate_ERP_preprocessing_IO.m [hunk 5]
```diff
@@ -75,326 +66,77 @@
...
-          
+				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{2} '_Corrected_Standard.set']
+				 %notice this is opposite the goofy
+			else
+				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
+            end
             
-        Filename = [subs{i_sub} '_' exp '_' conds{i_cond} '.vhdr'];
-                setname = Filename(1:end-5)
+        EEG = pop_selectevent( tempEEG, 'type',3 ,'renametype','Standard','deleteevents','on','deleteepochs','on','invertepochs','off');
```
Comment (kylemath) [inline Skate_ERP_preprocessing_IO.m]: make sure that this stuff is AFTER the checks above where the names are changed

File: Skate_ERP_preprocessing_IO.m [hunk 6]
```diff
@@ -75,326 +66,77 @@
...
+				ThisIsTheName = [subs{i_sub} '_' exp '_' new_conds{1} '_Corrected_Standard.set']
+            end
             
-        Filename = [subs{i_sub} '_' exp '_' conds{i_cond} '.vhdr'];
-                setname = Filename(1:end-5)
+        EEG = pop_selectevent( tempEEG, 'type',3 ,'renametype','Standard','deleteevents','on','deleteepochs','on','invertepochs','off');    
+        EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' new_conds{n_new} '_Corrected_Standard.set']);
+        EEG = pop_saveset( EEG, 'filename',ThisIsTheName,'filepath',[Pathname 'segments_IO_V2\']);
+
+%         EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' conds{i_cond} '_Corrected_Standard']);
```
Comment (kylemath) [inline Skate_ERP_preprocessing_IO.m]: DANIEL!!!! this line is specifically commented out, that is why it is not saving, come on!

**Comment by kylemath** [comment]
Moving these to a separate comments to make sure you address them:
---OTHER COMMENTS----
```
	Goofy_stance_info.txt
```
-this will be redudant after you add the vector of 1s and 0s to the script, and this info should be in the subNotes.txt file instead, so you can remove this file
```
    EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' conds{i_cond} '_Corrected_Standard']);
    EEG = pop_saveset( EEG, 'filename',[subs{i_sub} '_' exp '_facing_in_Corrected_Standard.set'],'filepath',[Pathname 'segmentsIO\']);
```
-you have to change the names in the first line too (editset), so they show up correctly in EEGLAB (these names are saved inside the .set file variables)
```
	if exist([Pathname 'segments\'])
	    mkdir([Pathname 'segmentsIO\']);
	end
```
-you need to change the name of both lines of the code,
```
	commandhistory
```
-What does this do?
```
        if i_cond ==1
        tempEEG1 =   EEG;
        elseif i_cond==2
        tempEEG2 = EEG;
```
-should be indented correctly after if and else
-everything between else and end should be indented

File: Skate_ERP_preprocessing_IO.m (hunk 4 above)
Comment (kylemath) [reply Skate_ERP_preprocessing_IO.m]: I'm not sure why you repeat the if elseif end statement twice, that just makes the name and you shouldn't have to run it for both targets and standards

File: Skate_ERP_preprocessing_IO.m (hunk 1 above)
Comment (d-robles) [reply Skate_ERP_preprocessing_IO.m]: I was following the example of the code.  It's easy to set it up the normal way we do it. That's what all the other codes are like. easy change.

File: Skate_ERP_preprocessing_IO.m (hunk 3 above)
Comment (d-robles) [reply Skate_ERP_preprocessing_IO.m:90]: okay

File: Skate_ERP_preprocessing_IO.m (hunk 6 above)
Comment (d-robles) [reply Skate_ERP_preprocessing_IO.m]: Lines 103, 104 and 105 have those same lines of code. That commented out piece of code is there to serve me as a reference.

File: Skate_ERP_preprocessing_IO.m (hunk 2 above)
Comment (d-robles) [reply Skate_ERP_preprocessing_IO.m:77]: I don't think so either

File: Skate_ERP_preprocessing_IO.m (hunk 4 above)
Comment (d-robles) [reply Skate_ERP_preprocessing_IO.m]: This part was confusing. 

The loop to change the names of the variables was identical to the loop we have on the behavioural file, so I can easily see how it's correctly assigning the conditions to the New_conds variable. 

But this is the ERP preprocessing, doesn't the loop need to be ran again to rename the conditions for the targets?

**Comment by d-robles** [comment]
The new version has all the loops properly aligned, and I fixed the issue saving the files. I had an end statement in the wrong spot. 

@kylemath

File: Skate_ERP_preprocessing_IO.m [hunk 7]
```diff
@@ -44,9 +35,9 @@
             EEG.data(x,:) = (EEG.data(x,:)-((EEG.data(EEG.nbchan-2,:))*.5));
         end
         
-        %Filter the data with low pass of 30
-               EEG = pop_eegfilt( EEG, .1, 0, [], 0);  %high pass filter
-               EEG = pop_eegfilt( EEG, 0, 30, [], 0);  %low pass filter
+%         %Filter the data with low pass of 30
```
Comment (d-robles) [inline Skate_ERP_preprocessing_IO.m]: The bandpass filter is still disabled. Makes testing faster. 

@kylemath

File: Skate_ERP_preprocessing_IO.m (hunk 4 above)
Comment (kylemath) [reply Skate_ERP_preprocessing_IO.m]: no since you can rename both the targets and standards inside the same loop

**Review by kylemath — APPROVED** [review]
sure this all looks fine to me, I guess the extra loop for targets and standards naming is fine, although you could still name them both in the same loop (just use different names), no big deal if it works

also I noticed you made a ERP_IO_V2.m, generally you don't want to do this, you want to edit the original file, or make a new file with a new purpose. Remember that github keeps track of all your changes. 

By making the new file V2, I can't see what you changed from V1, so you may want to rename V2 back to the original and delete the V2 version after this is merged

Well done on your first pull request review.

**Pull request merged**